script/install_icpc.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+#
+# Copyright 2020 Axel Huebl
+#
+# This file is part of alpaka.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+source ./script/travis_retry.sh
+
+source ./script/set.sh
+
+: "${CXX?'CXX must be specified'}"
+
+# Ref.: https://github.com/rscohn2/oneapi-ci
+# intel-basekit intel-hpckit are too large in size
+
+travis_retry sudo apt-get -qqq update
+travis_retry sudo apt-get install -y wget build-essential pkg-config cmake ca-certificates gnupg
+travis_retry sudo wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
+
+sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
+echo "deb https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
+
+travis_retry sudo apt-get update
+travis_retry sudo apt-get install -y intel-oneapi-icc
+
+set +eu
+source /opt/intel/inteloneapi/setvars.sh
+set -eu
+
+which "${CXX}"
+${CXX} -v
+which "${CC}"
+${CC} -v
```

Comment:
Could you try to add those 4 lines in front of the `./bootstrap.sh --with-toolset="intel-linux"`.
Maybe some environment vars get lost in between?

Reply:
I think so, yep.

Reply:
It may be necessary that you have to call `source /opt/intel/inteloneapi/setvars.sh` in `alpaka/script/run.sh` again.
I have not yet removed the split between the docker image creation (start, run install.sh, stop, save as image) and then building and executing the tests (start image, run.sh to build and test). The environment is not correctly preservert in between those steps.
I already started getting rid of this two step setup because we are not caching the docker images anymore with github actions.

Reply:
Looks like the `icc` is there. something funky in bootstrap of bjam is going on.

Reply:
Yep, the boost build system is archaic:
https://github.com/boostorg/build/issues/475